scripts/aws/manage_aws_stack.sh

```diff
@@ -0,0 +1,247 @@
+#!/bin/bash
+#
+# Script for management of an AWS stack.
+
+set -euo pipefail
+
+SCRIPTDIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+
+action_create() {
+    local -r stack_name="${1}"
+    local -r inst_type="${2}"
+    local -r region="${3}"
+    local -r pub_key="${4}"
+    local -r os="${5}"
+    local ami="${6}"
+
+    if [ -z "${stack_name}" ] ; then
+        echo "ERROR: stack name not set" 1>&2
+        exit 1
+    fi
+    if [ -z "${inst_type}" ] ; then
+        echo "ERROR: instance type not set" 1>&2
+        exit 1
+    fi
+    if [ -z "${region}" ] ; then
+        echo "ERROR: region type not set" 1>&2
+        exit 1
+    fi
+
+    local -r cf_tpl_file="${SCRIPTDIR}/cf-gen.yaml"
+    
+    local arch="x86_64"
+    if [[ "${inst_type%.*}" =~ .*"g".* ]]; then
+        arch="arm64"
+    fi
+    if [[ "${ami}" == "" ]]; then
+        ami=$(get_amis "${os}" "${arch}" "${region}" | head -n 1 | awk '{print $2}')
+    fi
+    local ec2Type="VirtualMachine"
+    if [[ "${inst_type}" =~ c[0-9]+[gn].metal ]]; then
+        ec2Type="MetalMachine"
+    fi
+    
+    echo "Stack name: ${stack_name}"
+    echo "Instance type: ${inst_type}" 
+    echo "OS: ${os}" 
+    echo "Arch: ${arch}" 
+    echo "AMI ID: ${ami}" 
+    echo "Region: ${region}"
+
+    aws --region "${region}" cloudformation create-stack --stack-name "${stack_name}" \
+        --template-body "file://${cf_tpl_file}" \
+        --capabilities CAPABILITY_NAMED_IAM \
+        --parameters \
+        ParameterKey=HostInstanceType,ParameterValue="${inst_type}" \
+        ParameterKey=Machinename,ParameterValue="${stack_name}" \
+        ParameterKey=AmiId,ParameterValue="${ami}" \
+        ParameterKey=EC2Type,ParameterValue="${ec2Type}" \
+        ParameterKey=PublicKeyString,ParameterValue="$(cat "${pub_key}")" \
+        ParameterKey=StackLaunchTemplate,ParameterValue="${stack_name}-launch-template"
+    
+    echo "Waiting for stack to be created"
+    aws --region "${region}" cloudformation wait stack-create-complete --stack-name "${stack_name}"
+    echo "Stack created successfully"
+    local instance_id
+    # shellcheck disable=SC2016
+    instance_id="$(aws --region "${region}" cloudformation describe-stacks --stack-name "${stack_name}" \
+        --query 'Stacks[].Outputs[?OutputKey == `InstanceId`].OutputValue' --output text)"
+    echo "Waiting for stack status to be OK"
+    aws --region "${region}" ec2 wait instance-status-ok --instance-id "${instance_id}"
+    echo "Stack status OK"
+    local public_ip
+    # shellcheck disable=SC2016
+    public_ip=$(aws --region "${region}" cloudformation describe-stacks --stack-name "${stack_name}" \
+        --query 'Stacks[].Outputs[?OutputKey == `PublicIp`].OutputValue' --output text)
```

Comment:
Another nit: just my personal visual preference to have distinct blocks visually obvious
```suggestion
    echo "Waiting for stack to be created"
    aws --region "${region}" cloudformation wait stack-create-complete --stack-name "${stack_name}"
    echo "Stack created successfully"

    local instance_id
    # shellcheck disable=SC2016
    instance_id="$(aws --region "${region}" cloudformation describe-stacks --stack-name "${stack_name}" \
        --query 'Stacks[].Outputs[?OutputKey == `InstanceId`].OutputValue' --output text)"

    echo "Waiting for stack status to be OK"
    aws --region "${region}" ec2 wait instance-status-ok --instance-id "${instance_id}"
    echo "Stack status OK"

    local public_ip
    # shellcheck disable=SC2016
    public_ip=$(aws --region "${region}" cloudformation describe-stacks --stack-name "${stack_name}" \
        --query 'Stacks[].Outputs[?OutputKey == `PublicIp`].OutputValue' --output text)
```

Reply:
fixed